Zend/zend_vm_def.h

```diff
@@ -1500,15 +1500,9 @@ ZEND_VM_HELPER(zend_pre_inc_helper, VAR|CV, ANY)
 			}
 		}
 		increment_function(var_ptr);
-		if (UNEXPECTED(EG(exception))) {
-			/* Smart branch expects result to be set with exceptions */
-			if (UNEXPECTED(RETURN_VALUE_USED(opline))) {
-				ZVAL_NULL(EX_VAR(opline->result.var));
-			}
-			HANDLE_EXCEPTION();
-		}
 	} while (0);
 
+	/* opcodes are expected to set the result value */
```

Comment:
Nit: This comment seems rather useless. I'd prefer just removing it. `result` is primarily assigned for when the result of `++$foo` is used. Nobody will understand the relation to the throw helper issue from this comment.